src/app/component/circular-heatmap/circular-heatmap.component.ts

```diff
@@ -58,124 +65,116 @@ export class CircularHeatmapComponent implements OnInit {
-              'Done%': -1,
-              Activity: [],
-            };
-            var totalTeamsImplemented: number = 0;
-            var totalActivityTeams: number = 0;
-            tempData['Dimension'] = allDimensionNames[i];
-            tempData['SubDimension'] = allSubDimensionInThisDimension[j];
-            tempData['Level'] = 'Level ' + (l + 1);
-            for (var k = 0; k < allActivityInThisSubDimension.length; k++) {
+            var level = 'Level ' + (l + 1);
+            var activity: activitySchema[] = [];
+            var activityCompletionStatus: number = -1;
+
+            for (var a = 0; a < allActivityInThisSubDimension.length; a++) {
               try {
                 var lvlOfCurrentActivity =
-                  this.YamlObject[allDimensionNames[i]][
-                    allSubDimensionInThisDimension[j]
-                  ][allActivityInThisSubDimension[k]]['level'];
-                if (lvlOfCurrentActivity == l + 1) {
-                  totalActivityTeams += 1;
-                  var nameOfActivity: string = allActivityInThisSubDimension[k];
+                  this.YamlObject[allDimensionNames[d]][
+                    allSubDimensionInThisDimension[s]
+                  ][allActivityInThisSubDimension[a]]['level'];
 
-                  // Create an object from an array from meta data
+                if (lvlOfCurrentActivity == l + 1) {
+                  var nameOfActivity: string = allActivityInThisSubDimension[a];
+                  var teamStatus: { [key: string]: boolean } = {};
                   const teams = this.teamList;
 
-                  var teamStatus: { [key: string]: boolean } = {};
+                  totalActivityTeams += 1;
 
                   teams.forEach((singleTeam: any) => {
                     teamStatus[singleTeam] = false;
                   });
 
                   var teamsImplemented: any =
-                    this.YamlObject[allDimensionNames[i]][
-                      allSubDimensionInThisDimension[j]
-                    ][allActivityInThisSubDimension[k]]['teamsImplemented'];
+                    this.YamlObject[allDimensionNames[d]][
+                      allSubDimensionInThisDimension[s]
+                    ][allActivityInThisSubDimension[a]]['teamsImplemented'];
+
                   if (teamsImplemented) {
                     teamStatus = teamsImplemented;
                   }
 
-                  // Calculating %done
+                  var localStorageData = this.getFromBrowserState();
+
+                  if (localStorageData != null && localStorageData.length > 0) {
+                    this.YamlObject[allDimensionNames[d]][
+                      allSubDimensionInThisDimension[s]
+                    ][allActivityInThisSubDimension[a]]['teamsImplemented'] =
+                      this.getTeamImplementedFromJson(
+                        localStorageData,
+                        allActivityInThisSubDimension[a]
+                      );
+                  }
```

Comment:
This is the main change everything else is refactoring, it keeps the Teams data in sync in the YAML Object